lib/howl/interactions/line_selection.moon

```diff
@@ -0,0 +1,97 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+import interact from howl
+import highlight from howl.ui
+import Matcher from howl.util
+
+line_match_highlighter = (editor) ->
+  local buffer
+  (selection, text, items) ->
+    if buffer
+      highlight.remove_all 'search', buffer
+      highlight.remove_all 'search_secondary', buffer
+
+    return unless selection
+
+    buffer = editor.buffer
+    line = buffer.lines[selection.line_nr]
+    local positions
+    if text and not text.is_empty
+      positions = Matcher.explain text, line.text
+
+    if positions
+      start_pos = line.start_pos
+      column = positions[1]
+      for hl_pos in *positions
+        highlight.apply 'search', buffer, start_pos + hl_pos - 1, 1
+    else
+      highlight.apply 'search', buffer, line.start_pos, line.end_pos - line.start_pos
+
+    -- highlight nearby secondary matches
+    if #items > 0 and not text.is_empty
+      local idx
+
+      for i, item in ipairs items
+        continue unless item.buffer == buffer
+        if item.line_nr >= editor.line_at_top - 5
+          idx = i
+          break
+
+      if idx
+        for i = idx, #items
+          continue unless items[i].buffer == buffer
+          nr = items[i].line_nr
+          if nr > editor.line_at_bottom + 5
+            break
+          if nr == selection.line_nr
+            continue
+          line = buffer.lines[nr]
+          start_pos = line.start_pos
+          positions = Matcher.explain text, line.text
+          if positions
+            for hl_pos in *positions
+              highlight.apply 'search_secondary', buffer, start_pos + hl_pos - 1, 1
+
+interact.register
+  name: 'select_line'
+  description: 'Selection for buffer lines'
+  handler: (opts) ->
+    lines = opts.lines
+    opts.lines = nil
+
+    unless lines
+      error '"lines" field required', 2
+
+    if opts.matcher or opts.items or opts.on_selection_change
+      error '"matcher", "items" or "on_selection_change" not allowed', 2
+
+    editor = opts.editor or howl.app.editor
+    opts = moon.copy opts
+    line_items = [{tostring(line.nr), line.chunk, buffer: line.buffer, line_nr: line.nr, :line} for line in *lines]
+
+    selected_line = opts.selected_line
+    if selected_line
+      for item in *line_items
+        if item.line_nr == selected_line.nr and item.buffer == selected_line.buffer
+          opts.selection = item
+          break
+      opts.selected_line = nil
+
+    matcher = Matcher line_items, preserve_order: true
+    opts.matcher = matcher
+    opts.on_selection_change = line_match_highlighter(editor)
+
+    result = interact.select_location opts
+
+    if result
+      line = result.selection.line
+      return {
+        :line
+        text: result.text
+        positions: matcher.explain result.text, line.text
```

Comment:
I'm thinking we could skip returning `positions` here, and just return `column` instead. From the code using this it's unclear how positions would be useful, since they're typically related to the interactive search. Once it's done it's done. The only possible case I could think of would be to keep the highlights, but that doesn't seem useful anyway. What do you say?